framework/src/partitioner/CopyMeshPartitioner.C

```diff
@@ -0,0 +1,218 @@
+
+  // First check that it makes sense to copy the partitioning
+  // If we use the same number of procs to partition this mesh than to partition the source mesh, we
+  // are effectively matching regions to processes the same way in both meshes. Makes sense.
+  // If we use more procs, this will leave some processes with no elements. It is not ideal, let's
+  // just give a warning. This does not happen in practice with MultiApps.
+  // If we use less procs (N_source > N_current), we error. If we could avoid erroring, then either:
+  // - the elements on our mesh always get matched to elements from only N_current processes among
+  //   the N_source processes. We can accomodate that
+  // - the elements on our mesh get matched to more processes than we have. We would
+  //   need a heuristic (for example a nested partitioner) to re-distribute these elements. We won't
+  //   support it
+  std::set<unsigned int> pids_used;
+
+  // We cannot get the point locator only on some ranks of the base mesh, so we must error here
+  if (_base_mesh->comm().size() > mesh.comm().size())
+    mooseError("This partitioner does not support using less ranks to partition the mesh than are "
+               "used to partition the source mesh (the mesh we are copying the partitioning from)");
+
+  // Get point locator
+  auto pl = _base_mesh->getPointLocator();
+  pl->enable_out_of_mesh_mode();
+
+  // We use the pull_parallel_vector_data algorithm.
+  // This code is very similar to partitioner::assign_partitioning in libMesh partitioner.C
+  // We need to define three things:
+  // - the list of requests: the elements in our local mesh we are trying to assign pids to
+  // - how to gather results: not all processors know which element belongs where, they will each
+  //   process the requests sent to them and send back the elem pids if they know them.
+  // - how to fill the requests: from the data gathered and sent back, we need to decide which data
+  //   is good and save it into the output map
+
+  // For each processor id, all the elements in our mesh that will request to know their new
+  // processor id. If the target mesh is replicated, that's all the elements from that mesh, but the
+  // source mesh may be distributed so we still need the communication. If a distributed mesh, then
+  // that's only the local and ghosted elements.
+  // We send the target element vertex average, enough to find it in the source mesh.
+  // We send the target element pid, to know where to save the result in the filled requests
+  // We send an index for the ordering of the requests to facilitate retrieving the results
+  // NOTE: in partitioner.C they manage to unpack the filled requests in the same order that
+  //       the requests are sent, thus saving the need for that last index. Possible rework..
+  std::map<processor_id_type,
+           std::vector<std::tuple<Real, Real, Real, processor_id_type, unsigned int>>>
```

Comment:
You can't do a `RealVectorValue`?

Reply:
Havent tried. Point did not work, hence the 3 Reals.
I can try

Reply:
lemme try. I did not include that one I had the two others